raft/raft.go

```diff
@@ -1122,9 +1122,15 @@ func stepLeader(r *raft, m pb.Message) error {
 		pr.RecentActive = true
 
 		if m.Reject {
-			r.logger.Debugf("%x received MsgAppResp(MsgApp was rejected, lastindex: %d) from %x for index %d",
-				r.id, m.RejectHint, m.From, m.Index)
+			r.logger.Debugf("%x received MsgAppResp(MsgApp was rejected, lastindex: %d) from %x for index %d, log term %d",
+				r.id, m.RejectHint, m.From, m.Index, m.LogTerm)
 			if pr.MaybeDecrTo(m.Index, m.RejectHint) {
```

Comment:
PS while we're here - the comment on `MaybeDecrTo` is wrong. It says that the first argument (`rejected`) is the index that we tried to append. But that's not true, we tried to append `rejected+1` (i.e. the `m.Index` of the MsgApp was `rejected`) and it failed due to a mismatch at `rejected` itself.